apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nav-sidebar/src/components/nav-sidebar/index.tsx

```diff
@@ -45,7 +49,10 @@ const Button = forwardRef(
 function WpcomBlockEditorNavSidebar() {
 	const { toggleSidebar, setSidebarClosing } = useDispatch( STORE_KEY );
 	const [ isOpen, isClosing, postType, selectedItemId, siteTitle ] = useSelect( ( select ) => {
-		const { getPostType, getSite } = select( 'core' ) as any;
+		const { getPostType, getSite } = ( select( 'core' ) as unknown ) as {
+			getPostType: ( postType: string ) => null | { slug: string };
+			getSite: () => null | { title: string };
```

Comment:
Retyping the result to what we use here. `select( 'core' )` wasn't giving any types.

Reply:
out of curiosity, what is `as unknown` doing here?

Reply:
I believe that's because that intermediary type assertion to `unknown` is necessary to assign the correct type. Without it, you'd get a TypeScript error saying

```
Conversion of type 'typeof import("/Users/marcociampini/Code/wp-calypso/node_modules/@types/wordpress__core-data/selectors")' to type '{ getPostType: (postType: string) => { slug: string; } | null; getSite: () => { title: string; } | null; }' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
```

Reply:
TIL, thanks!